torchgeo/trainers/segmentation.py

```diff
@@ -190,27 +199,112 @@ def configure_metrics(self) -> None:
         .. note::
            * 'Micro' averaging suits overall performance evaluation but may not reflect
              minority class accuracy.
-           * 'Macro' averaging, not used here, gives equal weight to each class, useful
+           * 'Macro' averaging gives equal weight to each class, useful
              for balanced performance assessment across imbalanced classes.
         """
         num_classes: int = self.hparams['num_classes']
         ignore_index: int | None = self.hparams['ignore_index']
-        metrics = MetricCollection(
-            [
-                MulticlassAccuracy(
+        labels: list[str] | None = self.hparams['labels']
+
+        self.train_metrics = MetricCollection(
```

Comment:
Can this be done in a for-loop?

Reply:
Something like the following? I think it is less readable 
```python
# Define the metrics and their parameters
metric_definitions = {
    'OverallAccuracy': (Accuracy, {'average': 'micro'}),
    'OverallF1Score': (FBetaScore, {'average': 'micro', 'beta': 1.0}),
    'OverallJaccardIndex': (JaccardIndex, {'average': 'micro'}),
    'AverageAccuracy': (Accuracy, {'average': 'macro'}),
    'AverageF1Score': (FBetaScore, {'average': 'macro', 'beta': 1.0}),
    'AverageJaccardIndex': (JaccardIndex, {'average': 'macro'}),
}

classwise_metric_definitions = {
    'Accuracy': Accuracy,
    'Precision': Precision,
    'Recall': Recall,
    'F1Score': (FBetaScore, {'beta': 1.0}),
    'JaccardIndex': JaccardIndex,
}

# Initialize the metrics dictionary
metrics_dict = {}

# Populate the metrics dictionary using the metric definitions
for name, (metric_class, params) in metric_definitions.items():
    metrics_dict[name] = metric_class(
        task='multiclass',
        num_classes=num_classes,
        multidim_average='global',
        ignore_index=ignore_index,
        **params
    )

for name, metric_class in classwise_metric_definitions.items():
    params = {}
    if isinstance(metric_class, tuple):
        metric_class, params = metric_class
    metrics_dict[name] = ClasswiseWrapper(
        metric_class(
            task='multiclass',
            num_classes=num_classes,
            average='none',
            multidim_average='global',
            ignore_index=ignore_index,
            **params
        ),
        labels=labels
    )

self.train_metrics = MetricCollection(metrics_dict, prefix='train_')
```

Reply:
Something like:
```python
for average in ['macro', 'micro', 'none']:
    ...
```
May need an if-statement for 'none'.
May need

Reply:
Th F1 score doesn't quite fit the pattern:
```python
from torchmetrics import MetricCollection, Accuracy, FBetaScore, JaccardIndex, Precision, Recall

# Define metric classes and their parameters
metric_classes = {
    'Accuracy': Accuracy,
    'F1Score': FBetaScore,
    'JaccardIndex': JaccardIndex,
    'Precision': Precision,
    'Recall': Recall,
}

# Define a dictionary to hold the metric instances
metrics_dict = {}

# Loop through the types of averaging
for average in ['micro', 'macro']:
    for metric_name, metric_class in metric_classes.items():
        name = f"Overall{metric_name}" if average == 'micro' else f"Average{metric_name}"
        params = {
            'task': 'multiclass',
            'num_classes': num_classes,
            'average': average,
            'multidim_average': 'global',
            'ignore_index': ignore_index,
        }
        if metric_name == 'F1Score':
            params['beta'] = 1.0
        metrics_dict[name] = metric_class(**params)

# Loop through the classwise metrics
for metric_name, metric_class in metric_classes.items():
    if metric_name != 'F1Score':
        metrics_dict[metric_name] = ClasswiseWrapper(
            metric_class(
                task='multiclass',
                num_classes=num_classes,
                average='none',
                multidim_average='global',
                ignore_index=ignore_index,
            ),
            labels=labels
        )
    else:
        metrics_dict[metric_name] = ClasswiseWrapper(
            metric_class(
                task='multiclass',
                num_classes=num_classes,
                beta=1.0,
                average='none',
                multidim_average='global',
                ignore_index=ignore_index,
            ),
            labels=labels
        )

# Create the MetricCollection with the prefix 'train_'
self.train_metrics = MetricCollection(metrics_dict, prefix='train_')
```

Reply:
Later we could refactor into a metrics factory method